src/index.ts

```diff
@@ -54,8 +56,10 @@ export = {
           }
         ],
         '@typescript-eslint/no-angle-bracket-type-assertion': 'error',
+        '@typescript-eslint/no-empty-function': 'error',
         '@typescript-eslint/no-empty-interface': 'error',
         '@typescript-eslint/no-extraneous-class': 'error',
+        '@typescript-eslint/no-floating-promises': 'error',
```

Comment:
I love where this is going, where so many rules are possible only when type information is available. This one, for example.